File: tools-excel/src/main/java/cn/gjing/tools/excel/EnumConvert.scala
package cn.gjing.tools.excel

/**
 * Enumeration converter
 * @author Gjing
 **/
trait EnumConvert[T <: Enum[_], E] {
  /**
   * Converted to entity enum field
   *
   * @param e Excel value
   * @return Entity value
   */
  def toEntityAttribute(e: E): T

  /**
   * Converted to excel value
   *
   * @param t Entity value
   * @return E Excel value
   */
  def toExcelAttribute(t: T): E
}

File: tools-excel/src/main/java/cn/gjing/tools/excel/convert/EnumConvert.java
package cn.gjing.tools.excel.convert;

/**
 * Enumeration converter
 *
 * @author Gjing
 **/
public interface EnumConvert<T extends Enum<?>, E> {
    /**
     * Converted to entity enum field
     *
     * @param e Excel cell value
     * @return enum
     */
    T toEntityAttribute(E e);

    /**
     * Converted to excel value
     *
     * @param t t Current field value
     * @return E
     */
    E toExcelAttribute(T t);

}
